Release/1.52.0 (#3192)
FusionAuth/fusionauth-site: astro/src/content/docs/release-notes/index.mdx

  Removal of this field is targeted for the end of 2024.

  **Docker architectures**

  We are planning to discontinue publishing Docker images for the following architectures: `linux/arm/v7`, `linux/ppc64le`, and `linux/s390x`. The rationale behind this decision is that we do not believe they are actively being used, and we would like to move to the GraalVM Java distribution which does not provide builds for these architectures.
   * https://hub.docker.com/r/fusionauth/fusionauth-app

  We plan to stop publishing docker images for these architectures at the end of 2024. If you are actively using any of these architectures, please let us know how this could affect you by contacting support or reaching out to sales.
</DeprecationWarning>

### Security

* When detecting impossible travel or similarly suspicious login events, it is possible that not all device trust cookies were correctly revoked.  These are now automatically revoked.
  * Resolves [GitHub Issue #2753](https://github.com/FusionAuth/fusionauth-issues/issues/2753)

### New

> * A free community license is now available, which adds WebAuthn (Passkeys) to the community edition. All those with a Community license will now find a license key in their [FusionAuth account plan](https://account.fusionauth.io/account/plan) page. And there was much rejoicing! 🥳

Reviewer: I don't mind them, but this release doubles the number of emoji we have in the release notes. Just wanted to call out the change in tone.

Author: Noted.

  * Resolves [GitHub Issue #2662](https://github.com/FusionAuth/fusionauth-issues/issues/2662)
  * Resolves [GitHub Issue #2663](https://github.com/FusionAuth/fusionauth-issues/issues/2663)

### Fixed
* Clicking the toggle checkbox element in the admin UI quickly may cause the checkbox state to be inverted. This can be easily fixed by refreshing the page. You should now be able to click as fast as you want! Go forth and click!
  * Resolves [GitHub Issue #2718](https://github.com/FusionAuth/fusionauth-issues/issues/2718)

* Attempting to sort API keys by key value in the admin UI by clicking the key value header would result in an error.
  * Resolves [GitHub Issue #2738](https://github.com/FusionAuth/fusionauth-issues/issues/2738)

* When using the API Key API and specifying an invalid `tenantId` on the request in order to create a tenant-scoped API key, the request fails with a `500` status code. This error has been corrected, and an appropriate validation error is now returned.
  * Resolves [GitHub Issue #2749](https://github.com/FusionAuth/fusionauth-issues/issues/2749)

> * The date picker component used in simple themes for things like birthdates was not styled correctly based upon the selected theme. The date picker has been changed to use the browser-default date picker which should work much better on mobile devices. This picker style will now be used in themed hosted login pages, as well as the admin UI for searching a date range or selecting a birthdate. This change should not affect any existing advanced theme that may still be using the older style date picker.

Reviewer 2: ```suggestion
* The date picker from PrimeJS, which was used for birthdates and custom date fields, was not styled correctly based upon the selected theme. The date picker has been changed to the browser-default date picker which should work much better on mobile devices. This picker style will now be used in themed hosted login pages, as well as the admin UI for searching a date range or selecting a birthdate. This change should not affect any existing advanced theme that may still use the older style date picker. See the updated `_helpers.ftl` file in the default theme to see the changes if you want to incorporate them into existing themes.
```

  * Resolves [GitHub Issue #2770](https://github.com/FusionAuth/fusionauth-issues/issues/2770)

* Adding custom message keys to your theme messages using the admin UI was failing to persist these changed messages. The UI for editing messages in the simple theme editor has also been improved to make it easier to understand which messages have been modified.
  * Resolves [GitHub Issue #2778](https://github.com/FusionAuth/fusionauth-issues/issues/2778)

> * The preview window for simple themes in the admin UI was showing un-styled pages when using the Firefox browser. Sorry Firefox users, we ask for your forgiveness. 😔

Reviewer 2: ```suggestion
* When the <InlineUIElement>Browser preview</InlineUIElement> button was used to open a new tab for simple themes in the admin UI the page would render without any applied CSS when using the Firefox browser. Sorry, Firefox users, we ask for your forgiveness. 😔
```
will need an import added if you accept this suggestion

  * Resolves [GitHub Issue #2794](https://github.com/FusionAuth/fusionauth-issues/issues/2794)

* The default `orderBy` parameter value for the [Group Member Search API](/docs/apis/groups#search-for-group-members) did not provide a consistent ordering of results because the default sort was on `insertInstant ASC` which may not always be unique. This API is used by the SCIM Groups Resource API which then can cause inconsistent results for the SCIM client. The default `orderBy` is now set to `insertInstant ASC, userId ASC, groupId ASC` to ensure a consistent result between API calls.
  * Resolves [GitHub Issue #2798](https://github.com/FusionAuth/fusionauth-issues/issues/2798)

> * When using the simple theme editor in the admin UI, the color picker did not always render next to the input field. The color picker will now always correctly render adjacent to the input field you select.

Reviewer 2: ```suggestion
* When using the simple theme editor in the admin UI the color picker did not always render next to the input field. The color picker should now correctly render adjacent to the appropriate input field you select.
```

`always` is technically untrue and out of our control as the browser gets to decide.

  * Resolves [GitHub Issue #2803](https://github.com/FusionAuth/fusionauth-issues/issues/2803)

> * Newlines and tabs were not being rendered when viewing audit entries in the view dialog from the admin UI. If you are using new lines or tabs in your audit log messages, you may not enjoy viewing them in all of their their intended glory!

Reviewer: ```suggestion
* Newlines and tabs were not being rendered when viewing audit entries in the view dialog from the admin UI. If you are using new lines or tabs in your audit log messages, you may now enjoy viewing them in all of their intended glory!
```

> * Newlines and tabs were not being rendered when viewing audit entries in the view dialog from the admin UI. If you are using new lines or tabs in your audit log messages, you may not enjoy viewing them in all of their their intended glory!

Reviewer 2: ```suggestion
* Newlines and tabs were not rendered when viewing audit entries in the view dialog from the admin UI. If you are using new lines or tabs in your audit log messages, you may now enjoy viewing them in all their intended glory!
```

  * Resolves [GitHub Issue #2808](https://github.com/FusionAuth/fusionauth-issues/issues/2808)

> * When using the interactive maintenance mode to upgrade your database schema, it is possible that you had to click the Submit button twice to exit maintenance mode. This was only a cosmetic issue but may be annoying of confusing to the user. We are sorry if you had to click the submit button twice. 😬

Reviewer: I think these should be consistent. Or just remove the second `Submit`.
```suggestion
* When using the interactive maintenance mode to upgrade your database schema, it is possible that you had to click the Submit button twice to exit maintenance mode. This was only a cosmetic issue but may be annoying of confusing to the user. We are sorry if you had to click the Submit button twice. 😬
```

> * When using the interactive maintenance mode to upgrade your database schema, it is possible that you had to click the Submit button twice to exit maintenance mode. This was only a cosmetic issue but may be annoying of confusing to the user. We are sorry if you had to click the submit button twice. 😬

Reviewer 2: ```suggestion
* When using the interactive maintenance mode to upgrade your database schema, it is possible that you had to click the Submit button twice to exit maintenance mode. This was only a cosmetic issue, however it may be annoying or confusing to the user. We are sorry if you had to click the submit button twice. 😬
```

  * Resolves [GitHub Issue #2815](https://github.com/FusionAuth/fusionauth-issues/issues/2815)

### Enhancements
* Add the new health check endpoint (`/api/health`) that was added in `1.51.1` to the client libraries.
  * Resolves [GitHub Issue #2804](https://github.com/FusionAuth/fusionauth-issues/issues/2804)

### Internal
* For users in FusionAuth Cloud, attempting to save a Simple theme may result in an error.
  * Resolves [GitHub Issue #2777](https://github.com/FusionAuth/fusionauth-issues/issues/2777)

> * An equals (`=`) sign in query parameter value was not being parsed correctly. There are no known issues related to this bug as generally speaking an equals (`=`) sign will be URL encoded as `%3D`. However because it is legal use an equals (`=`) sign un-encoded in a query string name or query string value, this has been corrected.

Reviewer: ```suggestion
* An equals (`=`) sign in query parameter value was not being parsed correctly. There are no known issues related to this bug as generally speaking an equals (`=`) sign will be URL encoded as `%3D`. However, because it is legal use an equals (`=`) sign un-encoded in a query string name or query string value, this has been corrected.
```

> * An equals (`=`) sign in query parameter value was not being parsed correctly. There are no known issues related to this bug as generally speaking an equals (`=`) sign will be URL encoded as `%3D`. However because it is legal use an equals (`=`) sign un-encoded in a query string name or query string value, this has been corrected.

Reviewer 2: ```suggestion
* An equals (`=`) sign in query parameter values was not being parsed correctly. There are no known issues related to this bug as generally speaking an equals (`=`) sign will be URL encoded as `%3D`. However, an unencoded equals (`=`) sign is illegal to use in a query string name or value. This has been corrected.
```

  * Resolves [GitHub Issue #2792](https://github.com/FusionAuth/fusionauth-issues/issues/2792)

> * An unused template was removed from the self-service login workflow. In practice this page was never rendered, and was not included in the theme configuration. This change should not impact anyone using themes.

Reviewer: The second half of this sentence does not contain an independent clause, so it is not a compound sentence and does not need a comma.
```suggestion
* An unused template was removed from the self-service login workflow. In practice this page was never rendered and was not included in the theme configuration. This change should not impact anyone using themes.
```

* When using the simple theme editor in the admin UI, the color picker did not always render next to the input field. The color picker will now always correctly render adjacent to the input field you select.
  * Resolves [GitHub Issue #2803](https://github.com/FusionAuth/fusionauth-issues/issues/2803)

> * Newlines and tabs were not rendered when viewing audit entries in the view dialog from the admin UI. If you are using new lines or tabs in your audit log messages, you may now enjoy viewing them in all of their intended glory!

Reviewer: IJ has taught me that including `of` in this case is redundant.
```suggestion
* Newlines and tabs were not rendered when viewing audit entries in the view dialog from the admin UI. If you are using new lines or tabs in your audit log messages, you may now enjoy viewing them in all their intended glory!
```

  * Resolves [GitHub Issue #2808](https://github.com/FusionAuth/fusionauth-issues/issues/2808)

> * When using the interactive maintenance mode to upgrade your database schema, it is possible that you had to click the Submit button twice to exit maintenance mode. This was only a cosmetic issue but may be annoying of confusing to the user. We are sorry if you had to click the Submit button twice. 😬

Reviewer: Should this use `<InlineUIElement>Submit</InlineUIElement>`?
